src/JokeraceEligibility.sol

```diff
@@ -134,53 +144,60 @@ contract JokeraceEligibility is HatsEligibilityModule {
 
-    if (currentContest.state() != IGovernor.ContestState.Completed) {
+    if (currentContest.state() != Governor.ContestState.Completed) {
       revert JokeraceEligibility_ContestNotCompleted();
     }
 
-    // sorted in ascending order
-    uint256[] memory sortedProposalIds = currentContest.sortedProposals(true);
-    uint256 numProposals = sortedProposalIds.length;
-    uint256 numEligibleWearers;
-
-    uint256 k = topK; // save SLOADs
-
-    // check if there's a tie between place k and k + 1. If so, election results are rejected
-    if (numProposals > k) {
-      numEligibleWearers = k;
-      // get the score of candidate in place K
-      uint256 placeK = numProposals - k; // only do this operation once
-      int256 totalVotesPlaceK = getTotalVotes(currentContest, sortedProposalIds[placeK]);
-      // get the score of candidate in place K + 1
-      int256 totalVotesPlaceKPlusOne = getTotalVotes(currentContest, sortedProposalIds[placeK - 1]);
-
-      if (totalVotesPlaceK == totalVotesPlaceKPlusOne) {
-        revert JokeraceEligibility_NoTies();
-      }
-    } else {
-      numEligibleWearers = numProposals;
-    }
-
-    for (uint256 i; i < numEligibleWearers;) {
-      address candidate = getCandidate(currentContest, sortedProposalIds[numProposals - i - 1]);
-      eligibleWearersPerContest[candidate][address(currentContest)] = true;
-
-      // should not overflow based on < numEligibleWearers stopping condition
-      unchecked {
-        ++i;
+    uint256 k = topK;
+    uint256 winningProposalsCount = 0;
+    uint256 currentRank = 1; // ranks start from '1' (the top-rank)
+    bool processed = false;
+    while (!processed) {
+      try currentContest.getRankIndex(currentRank) returns (uint256 rankIndex) {
+        // get the score of the curent rank (amount of 'for' votes)
+        uint256 forVotesOfCurrentRank = currentContest.sortedRanks(rankIndex);
+        // get the proposal IDs with the current score
+        uint256[] memory proposalsOfCurrentRank = currentContest.getProposalsWithThisManyForVotes(forVotesOfCurrentRank);
+        uint256 numProposalsOfCurrentRank = proposalsOfCurrentRank.length;
+        winningProposalsCount += numProposalsOfCurrentRank;
+
+        // if there's a tie
+        if (winningProposalsCount > k) {
+          termEnd = block.timestamp; // update the term end so that reelection will be immediately possible
+          revert JokeraceEligibility_NoTies();
+        }
+
+        // get the authors of the proposals and update their eligibility
+        for (uint256 i; i < numProposalsOfCurrentRank;) {
+          address candidate = getCandidate(currentContest, proposalsOfCurrentRank[i]);
+          eligibleWearersPerContest[candidate][address(currentContest)] = true;
+
+          unchecked {
+            ++i;
+          }
+        }
+
+        if (winningProposalsCount == k) {
+          processed = true;
+        }
+
+        currentRank += 1;
```

Comment:
can do this unchecked since k is never going to be high enough for it to overflow